bin/plugin/commands/test-plugins.js

```diff
@@ -0,0 +1,440 @@
+	}
+
+	// Validate that the plugins JSON file contains content before proceeding.
+	if (
+		'' === pluginsJsonFileContent ||
+		! pluginsJsonFileContent
+	) {
+		log(
+			formats.error(
+				`Contents of file at "${ settings.pluginsJsonFile }" could not be read, or are empty.`
+			)
+		);
+	}
+
+	const pluginsJsonFileContentAsJson = JSON.parse( pluginsJsonFileContent );
+
+	// Check for valid and not empty object resulting from plugins JSON file parse.
+	if (
+		'object' !== typeof pluginsJsonFileContentAsJson ||
+		0 === Object.keys( pluginsJsonFileContentAsJson ).length
+	) {
+		log(
+			formats.error(
+				`File at "settings.pluginsJsonFile" parsed, but detected empty/non valid JSON object.`
+			)
+		);
+
+		// Return with exit code 1 to trigger a failure in the test pipeline.
+		process.exit( 1 );
+	}
+
+	// Create an array of plugins from entries in plugins JSON file.
+	builtPlugins = Object.keys( pluginsJsonFileContentAsJson )
+		.filter( ( item ) => {
+			if ( ! fs.pathExistsSync( `${ settings.builtPluginsDir }${ pluginsJsonFileContentAsJson[ item ].slug }` ) ) {
+				log(
+					formats.error(
+						`Built plugin path "${ settings.builtPluginsDir }${ pluginsJsonFileContentAsJson[ item ].slug }" not found, skipping and removing from plugin list`
+					)
+				);
+				return false;
+			}
+			return true;
+		} )
+		.map( ( item ) => pluginsJsonFileContentAsJson[ item ].slug );
+
+	// For each built plugin, copy the test assets.
+	builtPlugins.forEach( ( plugin ) => {
+		log(
+			formats.success(
+				`Detected built plugin "${ plugin }", copying test files for standalone/integration testing.`
+			)
+		);
+
+		// Copy over test files.
+		try {
+			fs.copySync( settings.pluginTestAssets, `${ settings.builtPluginsDir }${ plugin }/`, {
+				overwrite: true,
+			} );
+			log(
+				formats.success(
+					`Copied test assets for plugin "${ plugin }", executing "composer install --no-interaction" on plugin.\n`
+				)
+			);
+		} catch ( e ) {
+			log(
+				formats.error(
+					`Error copying test assets for plugin "${ plugin }". ${ e }`
+				)
+			);
+
+			// Return with exit code 1 to trigger a failure in the test pipeline.
+			process.exit( 1 );
+		}
+
+		// Execute composer install within built plugin following copy.
+		execSync(
+			`composer install --working-dir=${ settings.builtPluginsDir }${ plugin } --no-interaction`,
+			( err, output ) => {
+				if ( err ) {
+					log( formats.error( `${ err }` ) );
+					process.exit( 1 );
+				}
+				// log the output received from the command
+				log( output );
+			}
+		);
+	} );
+
+	// Handle replacement of wp-env file content for round 1 of testing without root plugin.
+	doReplaceWpEnvContent( { ...settings, builtPlugins } );
+
+	// Run unit tests against built plugins.
+	doRunUnitTests( { ...settings, builtPlugins } );
+
+	log(
+		formats.success(
+			`Re-running standalone plugin integration tests with WPP plugin active.`
+		)
+	);
+	// Add the root level WPP plugin to the built plugins array.
```

Comment:
This works, but we could probably speed this up a bit by doing the first run with the WPP plugin active and then using a WP CLI command to disable it for the second run. That way we don't have to worry about restarting the environment.

Reply:
Great suggestion, I am in the process of changing this.